### PR TITLE
WIP: move from FactCheck to Base.Test

### PR DESCRIPTION
**Pull request by mlubin**
Still in progress.

@IainNZ, this counts as gardening

### CODE REVIEW

**Comment by mlubin** [comment]
This is basically done. Last missing piece is figuring out how to make it report the correct error status if a test fails (a la ``FactCheck.exitstatus()``).
@tkelman

**Comment by anriseth** [comment]
What is the reason for the move away from FactCheck?

**Comment by mlubin** [comment]
It is deprecated by Base.Test and it's ugly, have a peek at some of the new tests.

**Comment by mlubin** [comment]
Passing on 0.5, will look into failures on nightly later.

File: test/solvers.jl [hunk 1]
```diff
@@ -49,7 +49,7 @@ cbc && push!(lp_solvers, Clp.ClpSolver())
 glp && push!(lp_solvers, GLPKMathProgInterface.GLPKSolverLP())
 ipt && push!(lp_solvers, Ipopt.IpoptSolver(print_level=0))
 eco && push!(lp_solvers, ECOS.ECOSSolver(verbose=false))
-osl && push!(lp_solvers, CoinOptServices.OsilSolver())
+#osl && push!(lp_solvers, CoinOptServices.OsilSolver())
```
Comment (tkelman) [inline test/solvers.jl]: this probably shouldn't have been buried in a 5k line otherwise-unrelated commit

Comment (mlubin) [inline test/solvers.jl]: Oops
